tests/unit/valgrind.suppressions

```diff
@@ -29,11 +29,68 @@
    fun:main
 }
 
-# TODO: fix the pedantic leak errors from s2n_self_talk_alpn_test
+# START: s2n_self_talk_alpn_test suppressions
+# https://github.com/aws/s2n-tls/issues/1600
 {
    ignore_s2n_self_talk_alpn_test
    Memcheck:Leak
    match-leak-kinds: reachable
    ...
+   fun:s2n_cert_chain_and_key_new
    fun:main
```

Comment:
I suppose this is better than just "main", but this still applies to tests outside of s2n_self_talk_alpn_test, right?

Reply:
Correct. I'll go ahead and just fix the s2n_self_talk_alpn_test in this PR, that way we can get rid of all the suppressions.